### ISSUE TITLE
Import Feature Not Working

### ISSUE THREAD

**Issue by tango2590** [OttPeterR/addon-babybuddy#42]
It's not possible to even access the Import screen for any category. It results in a `404: Not Found` error. I specifically tried Feeding, Diaper Change, Pumping, and Sleep. I can access the export screens to save the data, but when I go to import it again, I can't do it through the add-on. It appears to be specific to the add-on, since I can access it correctly via the IP/port.

**Comment by OttPeterR** [comment]
This repository doesn't handle the source code of the actual app, that's over here: https://github.com/babybuddy/babybuddy

Edit: for extra context, the addon doesn't change the baby buddy app code or functionality. Are you able to do the import/export through the IP:port?

Edit 2: Okay I think I see what you're saying, you can't access the import database admin page through the ingress, but the ip:port works fine - is that right?

**Comment by tango2590** [comment]
> Edit 2: Okay I think I see what you're saying, you can't access the import database admin page through the ingress, but the ip:port works fine - is that right?

That is correct (and a better way of putting it, sorry). Attempting to access the import screen via ingress doesn't work, but I am able to access it by going directly to the IP:Port.

**Comment by OttPeterR** [comment]
I'm not able to replicate this on my end unfortunately, can you check the add-on logs and see if it's showing anything that might help me debug this? Like a 404 or something. Also some info about your setup may help - like architecture and what browser you're using.
For the time being, are you able to do your import/export stuff through the ip:port?

**Comment by tango2590** [comment]
Sorry for the delay. I can definitely use the import feature through the IP:Port when I need it. I tried getting some logs but instead of giving me an error, it just doesn't give me anything. I've pasted below a chunk of the log, starting with going into the Database Admin -> Feeding -> Export. It shows the correct GET call (`GET /admin/core/feeding/export/`). I then clicked "Feeding" in the breadcrumb menu and clicked Import, which resulted in nothing in the log. The actual HA page just says `404: Not Found` in the upper left of a black screen, but nothing in the add-on log.. Hope this helps.

As a side note, I wonder if all the "Closing connection"s have to do with the flash the screen does when you first open it. 

```
[2023-04-16 11:44:12 -0500] [254] [DEBUG] GET /admin/
[2023-04-16 11:44:13 -0500] [254] [DEBUG] Closing connection.
[2023-04-16 11:44:26 -0500] [255] [DEBUG] GET /admin/core/feeding/
[2023-04-16 11:44:27 -0500] [255] [DEBUG] Closing connection.
[2023-04-16 11:44:27 -0500] [255] [DEBUG] GET /admin/jsi18n/
[2023-04-16 11:44:27 -0500] [255] [DEBUG] Closing connection.
[2023-04-16 11:44:36 -0500] [254] [DEBUG] GET /admin/core/feeding/export/
[2023-04-16 11:44:37 -0500] [254] [DEBUG] Closing connection.
[2023-04-16 11:44:37 -0500] [255] [DEBUG] GET /static/admin/css/forms.c192d1ec6902.css
[2023-04-16 11:44:37 -0500] [255] [DEBUG] Closing connection.
[2023-04-16 11:44:37 -0500] [254] [DEBUG] GET /static/admin/css/widgets.00318bc424d3.css
[2023-04-16 11:44:37 -0500] [254] [DEBUG] Closing connection.
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/children/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/bmi/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/changes/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/feedings/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/head-circumference/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/height/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/notes/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/pumping/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/sleep/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/temperature/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/timers/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/tummy-times/
[2023-04-16 11:44:38 -0500] [254] [DEBUG] GET /api/weight/
[2023-04-16 11:44:50 -0500] [255] [DEBUG] GET /admin/core/feeding/
[2023-04-16 11:44:51 -0500] [255] [DEBUG] Closing connection.
[2023-04-16 11:44:51 -0500] [255] [DEBUG] GET /admin/jsi18n/
[2023-04-16 11:44:51 -0500] [255] [DEBUG] Closing connection.
```

**Comment by rodrihawk28** [comment]
Hi all, 

I'm in the exact same situation and my logs don't provide any info at all either. 

Some extra info that might not be related at all:
- The baby buddy addon uses more than 1GB of RAM simply by being running. Not a problem for me but it seems strange that it needs so much RAM. Could it be an indication of a different problem?
- When the addon is started, the last line of the log is always "starting NGINX":

```
[2023-08-02 10:55:21 +0100] [206] [INFO] Starting gunicorn 21.2.0
[2023-08-02 10:55:21 +0100] [206] [DEBUG] Arbiter booted
[2023-08-02 10:55:21 +0100] [206] [INFO] Listening at: http://0.0.0.0:8000 (206)
[2023-08-02 10:55:21 +0100] [206] [INFO] Using worker: gthread
[2023-08-02 10:55:21 +0100] [249] [INFO] Booting worker with pid: 249
[2023-08-02 10:55:21 +0100] [250] [INFO] Booting worker with pid: 250
[2023-08-02 10:55:21 +0100] [206] [DEBUG] 2 workers
[10:55:21] INFO: Starting NGINX...
[2023-08-02 09:55:21 +0000] [249] [DEBUG] Ignored premature client disconnection. No more data after: b'\n'
```

Let me know if I can be helpful in any way.

**Comment by OttPeterR** [comment]
Just to clarify, you mean the same situation where you can access the import screen through IP:port but it still does not import your data properly and shows similar logs to tango2590 when you do try the import feature?

That high ram usage might be coming from the data import... mine sits around 180Mb so over 1Gb is a lot. Could you try restarting the addon to see if that lowers your usage? I wonder if a memory limit or file upload size is the issue here. I have a feeling file i/o through ingress could be strange, but not sure.

The Nginx part is expected, the addon is running baby Buddy internally and it's using nginx to map the network connections for ingress and whichever port the user chooses for external access.

**Comment by rodrihawk28** [comment]
Ah, wait. I misunderstood the initial problem, but I was able to import now with your explanation.
See, the port number option is disabled by default when you install the addon. So I never really bothered to actually configure it:
![image](https://github.com/OttPeterR/addon-babybuddy/assets/63472280/c5d02bf4-9a8d-4717-ac81-1274576755e4)

I have configured it after your last reply and I was able to import through IP:port. I Imported 1700 entries at once so a high number is not the issue here.

I can definitely live with visiting IP:port for the imports so all good for me. Still, let me know if I can help in any way with this issue.  (Logs, testing, etc.)

I will create a new issue for the high RAM usage as it is unrelated to the import feature.
